offer-manager-domain/src/main/groovy/life/qbic/business/offers/Offer.groovy

```diff
@@ -279,15 +279,15 @@ class Offer {
     }
 
     private BigDecimal calculateItemNet(ProductItem item) {
-        return getUnitPrice(item) * item.quantity
+        return calculateUnitPrice(item) * item.quantity
     }
 
     private double calculateItemOverhead(ProductItem item) {
         return (calculateItemNet(item) - item.quantityDiscount) * overhead
     }
 
-    private BigDecimal getUnitPrice(ProductItem item){
-        return selectedCustomerAffiliation.category == AffiliationCategory.INTERNAL ? item.product.internalUnitPrice : item.product.externalUnitPrice
+    private BigDecimal calculateUnitPrice(ProductItem item){
+        return BigDecimal.valueOf(selectedCustomerAffiliation.category == AffiliationCategory.INTERNAL ? item.product.internalUnitPrice : item.product.externalUnitPrice)
```

Comment:
I think that variables with names make it more readable what is done here in the future. ```suggestion
boolean isInternalPrice = selectedCustomerAffiliation.category == AffiliationCategory.INTERNAL
BigDecimal unitPrice = BigDecimal.valueOf(isInternalPrice ? item.product.internalUnitPrice : item.product.externalUnitPrice)
return unitPrice
```

Reply:
Thanks for pointing that out, I would keep it as is for now